### PR TITLE
fix(Issue): Calculate first_response_time based on working hours

### PR DESCRIPTION
**Pull request by GangaManoj**
_Problem:_
First Response Time calculation does not take working hours into consideration- it just takes the difference between first_responded_on and time of creation of the Issue.

_Fix:_
Checks if first_responded_on and time of creation of the Issue are:

- on the same day
- during working hours
- on working days

### CODE REVIEW

File: erpnext/support/doctype/issue/issue.py [hunk 1]
```diff
@@ -212,7 +212,143 @@ def make_issue_from_communication(communication, ignore_communication_links=Fals
...
 
+def get_time_in_timedelta(time):
+	"""
+		Converts datetime.time(10, 36, 55, 961454) to datetime.timedelta(seconds=38215)
+	"""
+	return timedelta(hours=time.hour, minutes=time.minute, seconds=time.second)
+
+def set_first_response_time(communication, method):
+	parent = get_parent_doc(communication)
+	if parent.doctype == "Issue":
```
Comment (nextchamp-saqib) [inline erpnext/support/doctype/issue/issue.py]:
```suggestion
	if communicaton.get('reference_doctype') == "Issue":
		parent = get_parent_doc(communication)
```

Comment (nextchamp-saqib) [reply erpnext/support/doctype/issue/issue.py]: On every communication, we can prevent a `get_doc` to be executed

File: erpnext/support/doctype/issue/issue.py [hunk 2]
```diff
@@ -24,7 +24,7 @@ def validate(self):
 		if not self.raised_by:
 			self.raised_by = frappe.session.user
 
-		self.set_lead_contact(self.raised_by)
+		self.set_lead_contact(self.raised_by)
```
Comment (nextchamp-saqib) [inline erpnext/support/doctype/issue/issue.py]:
```suggestion
		self.set_lead_contact(self.raised_by)
```

**Review by nextchamp-saqib — CHANGES_REQUESTED** [review]

File: erpnext/support/doctype/issue/issue.py (hunk 2 above)
Comment (GangaManoj) [reply erpnext/support/doctype/issue/issue.py]: Done.

File: erpnext/support/doctype/issue/issue.py (hunk 1 above)
Comment (GangaManoj) [reply erpnext/support/doctype/issue/issue.py]: That sounds much better, I'll do that.

File: erpnext/support/doctype/issue/issue.py [hunk 3]
```diff
@@ -212,6 +212,128 @@ def make_issue_from_communication(communication, ignore_communication_links=Fals
...
+
+def set_first_response_time(communication, method):
+	if communication.get('reference_doctype') == "Issue":
+		issue = get_parent_doc(communication)
+		if is_first_response(issue):
+			first_response_time = calculate_first_response_time(issue, get_datetime(issue.first_responded_on))
+			issue.db_set("first_response_time", first_response_time)
+
+def is_first_response(issue):
+	responses = frappe.get_all('Communication', filters = {'reference_name': issue.name})
```
Comment (nextchamp-saqib) [inline erpnext/support/doctype/issue/issue.py]:
```suggestion
	responses = frappe.get_all('Communication', filters = {'reference_type': 'Issue', 'reference_name': issue.name, 'sent_or_received': 'Sent'})
```

**Review by nextchamp-saqib — CHANGES_REQUESTED** [review]

File: erpnext/support/doctype/issue/issue.py [hunk 4]
```diff
@@ -212,7 +212,129 @@ def make_issue_from_communication(communication, ignore_communication_links=Fals
...
+
+def set_first_response_time(communication, method):
+	if communication.get('reference_doctype') == "Issue":
+		issue = get_parent_doc(communication)
+		if is_first_response(issue):
+			first_response_time = calculate_first_response_time(issue, get_datetime(issue.first_responded_on))
+			issue.db_set("first_response_time", first_response_time)
+
+def is_first_response(issue):
+	responses = frappe.get_all('Communication', filters = {'reference_type': 'Issue', 'reference_name': issue.name, 'sent_or_received': 'Sent'})
```
Comment (nextchamp-saqib) [inline erpnext/support/doctype/issue/issue.py]:
```suggestion
	responses = frappe.get_all('Communication', filters = {'reference_name': issue.name, 'sent_or_received': 'Sent'})
```

**Review by nextchamp-saqib — CHANGES_REQUESTED** [review]

**Pull request merged**